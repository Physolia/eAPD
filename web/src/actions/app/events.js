import axios from '../../util/api';
import { selectApdData } from '../../reducers/apd.selectors';

export const saveApdEvent =
  (eventType, metadata = null) =>
  (dispatch, getState) => {
    const state = getState();
    const { id: apdID } = selectApdData(state);

    return axios
      .post(`/apds/${apdID}/events`, { eventType, metadata })
      .then(res => (res.data ? res.data : null));
  };

// Stryker disable next-line ObjectLiteral
export default { saveApdEvent };
