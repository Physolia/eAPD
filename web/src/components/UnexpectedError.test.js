import { shallow } from 'enzyme';
import React from 'react';

import {
  plain as UnexpectedError,
  mapStateToProps,
  mapDispatchToProps
} from './UnexpectedError';
import { saveApd } from '../actions/app';

describe('the unexpected error alert component', () => {
  it('renders as expected if there is no error and not saving', () => {
    expect(
      shallow(
        <UnexpectedError hasError={false} isSaving={false} save={jest.fn()} />
      )
    ).toMatchSnapshot();
  });

  it('renders as expected if there is an error and not saving', () => {
    expect(
      shallow(<UnexpectedError hasError isSaving={false} save={jest.fn()} />)
    ).toMatchSnapshot();
  });

  it('renders as expected if there is and error and is saving', () => {
    expect(
      shallow(<UnexpectedError hasError isSaving save={jest.fn()} />)
    ).toMatchSnapshot();
  });

  it('maps redux state to component props', () => {
    expect(
      mapStateToProps({
        saving: {
          error: 'this is the error',
          saving: 'this is whether we are saving'
        }
      })
    ).toEqual({
      hasError: true,
      isSaving: 'this is whether we are saving'
    });
  });

  it('maps redux actions to component props', () => {
    expect(mapDispatchToProps).toEqual({ save: saveApd });
  });
});
