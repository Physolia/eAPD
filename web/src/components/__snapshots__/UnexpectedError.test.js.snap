// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`the unexpected error alert component renders as expected if there is an error and not saving 1`] = `
<div
  aria-live="polite"
>
  <Dialog
    ariaCloseLabel="Close modal dialog"
    className="ds-u-left-border-warning"
    closeButtonText="Close"
    closeButtonVariation="transparent"
    closeIcon={<CloseIcon />}
    escapeExitDisabled={false}
    escapeExits={true}
    heading={
      <div>
        <FontAwesomeIcon
          border={false}
          className=""
          fixedWidth={false}
          flip={null}
          icon={
            Object {
              "icon": Array [
                576,
                512,
                Array [],
                "f071",
                "M569.517 440.013C587.975 472.007 564.806 512 527.94 512H48.054c-36.937 0-59.999-40.055-41.577-71.987L246.423 23.985c18.467-32.009 64.72-31.951 83.154 0l239.94 416.028zM288 354c-25.405 0-46 20.595-46 46s20.595 46 46 46 46-20.595 46-46-20.595-46-46-46zm-43.673-165.346l7.418 136c.347 6.364 5.609 11.346 11.982 11.346h48.546c6.373 0 11.635-4.982 11.982-11.346l7.418-136c.375-6.874-5.098-12.654-11.982-12.654h-63.383c-6.884 0-12.356 5.78-11.981 12.654z",
              ],
              "iconName": "exclamation-triangle",
              "prefix": "fas",
            }
          }
          inverse={false}
          listItem={false}
          mask={null}
          pull={null}
          pulse={false}
          rotation={null}
          size={null}
          spin={false}
          swapOpacity={false}
          symbol={false}
          title=""
          transform={null}
        />
         Unable to save changes!
      </div>
    }
    onExit={[Function]}
    underlayClickExits={false}
  >
    <p>
      Your changes aren't being saved. Verify your internet connection and try saving your changes again in a few minutes by clicking on the save button. If you refresh your browser without saving, changes made after this message appeared will be lost.
    </p>
    <p>
      If your internet connection is working and this message persists for more than 5 minutes, please contact us at
       
      <a
        href="mailto:CMS-EAPD@cms.hhs.gov"
      >
        CMS-EAPD@cms.hhs.gov
      </a>
      .
    </p>
    <p
      className="ds-u-text-align--right ds-u-margin-bottom--0"
    >
      <Button
        component="button"
        onClick={[MockFunction]}
        type="button"
        variation="primary"
      >
        Save
      </Button>
    </p>
  </Dialog>
</div>
`;

exports[`the unexpected error alert component renders as expected if there is and error and is saving 1`] = `
<div
  aria-live="polite"
>
  <Dialog
    ariaCloseLabel="Close modal dialog"
    className="ds-u-left-border-warning"
    closeButtonText="Close"
    closeButtonVariation="transparent"
    closeIcon={<CloseIcon />}
    escapeExitDisabled={false}
    escapeExits={true}
    heading={
      <div>
        <FontAwesomeIcon
          border={false}
          className=""
          fixedWidth={false}
          flip={null}
          icon={
            Object {
              "icon": Array [
                576,
                512,
                Array [],
                "f071",
                "M569.517 440.013C587.975 472.007 564.806 512 527.94 512H48.054c-36.937 0-59.999-40.055-41.577-71.987L246.423 23.985c18.467-32.009 64.72-31.951 83.154 0l239.94 416.028zM288 354c-25.405 0-46 20.595-46 46s20.595 46 46 46 46-20.595 46-46-20.595-46-46-46zm-43.673-165.346l7.418 136c.347 6.364 5.609 11.346 11.982 11.346h48.546c6.373 0 11.635-4.982 11.982-11.346l7.418-136c.375-6.874-5.098-12.654-11.982-12.654h-63.383c-6.884 0-12.356 5.78-11.981 12.654z",
              ],
              "iconName": "exclamation-triangle",
              "prefix": "fas",
            }
          }
          inverse={false}
          listItem={false}
          mask={null}
          pull={null}
          pulse={false}
          rotation={null}
          size={null}
          spin={false}
          swapOpacity={false}
          symbol={false}
          title=""
          transform={null}
        />
         Unable to save changes!
      </div>
    }
    onExit={[Function]}
    underlayClickExits={false}
  >
    <p>
      Your changes aren't being saved. Verify your internet connection and try saving your changes again in a few minutes by clicking on the save button. If you refresh your browser without saving, changes made after this message appeared will be lost.
    </p>
    <p>
      If your internet connection is working and this message persists for more than 5 minutes, please contact us at
       
      <a
        href="mailto:CMS-EAPD@cms.hhs.gov"
      >
        CMS-EAPD@cms.hhs.gov
      </a>
      .
    </p>
    <p
      className="ds-u-text-align--right ds-u-margin-bottom--0"
    >
      <Button
        component="button"
        onClick={[MockFunction]}
        type="button"
        variation="primary"
      >
        <Spinner />
         Saving
      </Button>
    </p>
  </Dialog>
</div>
`;

exports[`the unexpected error alert component renders as expected if there is no error and not saving 1`] = `
<div
  aria-live="polite"
/>
`;
